swarm/src/connection.rs

```diff
@@ -131,10 +131,10 @@ where
 
     /// Begins an orderly shutdown of the connection, returning the connection
     /// handler and a `Future` that resolves when connection shutdown is complete.
-    pub fn close(self) -> (THandler, impl Future<Output = io::Result<()>>) {
+    pub fn close(mut self) -> (THandler, impl Future<Output = io::Result<()>>) {
         (
             self.handler.into_connection_handler(),
-            poll_fn(move |cx| self.muxing.poll_close(cx)),
+            poll_fn(move |cx| self.muxing.poll_close_unpin(cx)),
```

Comment:
```suggestion
            self.muxing.close(),
```
Can we use the `StreamExt::close` here?

Reply:
I tried but unfortunately, we are moving `self` here and `close` doesn't move `self`. It could be made to move it but I am not sure if that isn't too restrictive?

I guess there isn't really a use for accessing the muxer after closing so we can move it into the future.

Reply:
I've changed `StreamMuxerExt::close` to consume `self` which allows us to use it here.